haas/cli.py

```diff
@@ -231,34 +231,29 @@ def object_url(*args):
     return url
 
 
-def do_request(method, url, data={}):
-    """Helper function for making HTTP requests against the API.
-
-    Uses the global variable `http_client` to make the request.
-
-    Arguments:
-
-        `method` - the http method, as a string: 'GET', 'PUT', 'POST'...
-        `url` - The url to make the request to
-        `data` - the body of the request.
-    """
-    return check_status_code(http_client.request(method, url, data=data))
-
+# Helper functions for making HTTP requests against the API.
+#    Uses the global variable `http_client` to make the request.
+#
+#    Arguments:
+#
+#        `url` - The url to make the request to
+#        `data` - the body of the request (for PUT, POST and DELETE)
+#        `params` - query parameters (for GET)
 
-def do_put(url, data={}):
-    return do_request('PUT', url, data=json.dumps(data))
+def do_put(url, data=None):
+    check_status_code(http_client.request('PUT', url, data=data))
```

Comment:
Do we need the json.dumps() here?

Jethro - could you look into whether the requests and keystone -middleware libraries will automatically convert to json?